src/xmlsec/xmldsig.rs

```diff
@@ -141,6 +203,32 @@ impl XmlSecSignatureContext {
         self.verify_node_raw(signode)
     }
 
+    /// This is to help with testing, but it attempts to verify all of the
+    /// signatures within a document and returns the number of verified
+    /// signatures.
+    ///
+    /// you should call `update_document_id_hash` before calling this if you
+    /// need to verify the ID hash.
+    pub fn verify_all_signatures(&self, doc: &XmlDocument) -> XmlSecResult<i32> {
+        self.key_is_set()?;
+        // Creating an XPath to locate all of the
+        let xpath_context = libxml::xpath::Context::new(doc).expect("Failed to create XPath");
+        xpath_context
+            .register_namespace("dsig", "http://www.w3.org/2000/09/xmldsig#")
+            .map_err(|_| XmlSecError::XPathNamespaceError)?;
```

Comment:
Could we preserve the error source here too?

Reply:
Nope. See other comment.